plotly/io/_base_renderers.py

```diff
@@ -637,3 +639,21 @@ def render(self, fig_dict):
             validate=False,
         )
         open_html_in_browser(html, self.using, self.new, self.autoraise)
+
+
+class SphinxGalleryRenderer(ExternalRenderer):
+
+    def render(self, fig_dict):
+        stack = inspect.stack()
+        # Name of script from which plot function was called is retrieved
+        try:
+            filename = stack[3].filename # let's hope this is robust...
+        except: #python 2
+            filename = stack[3][1]
+        filename_root, _ = os.path.splitext(filename)
+        filename_html = filename_root + '.html'
+        filename_png = filename_root + '.png'
+        figure = return_figure_from_figure_or_data(fig_dict, True)
+        _ = plot(fig_dict, auto_open=False,
```

Comment:
Let's change this from `plotly.offline.plot` to `plotly.io.write_html`, since that is the recommended way to write figures to HTML now.